core/app/status/logger.go

```diff
@@ -34,19 +35,26 @@ func RegisterLogger(progressUpdateFreq time.Duration) Unregister {
 }
 
 type statusLogger struct {
-	lastProgressUpdate map[*Task]time.Time
-	progressUpdateFreq time.Duration
+	lastProgressUpdate     map[*Task]time.Time
+	lastProgressUpdateLock sync.Mutex
+	progressUpdateFreq     time.Duration
 }
 
 func (l statusLogger) OnTaskStart(ctx context.Context, t *Task) {
 	log.I(ctx, "%v Started", t)
+	l.lastProgressUpdateLock.Lock()
 	l.lastProgressUpdate[t] = time.Now()
+	l.lastProgressUpdateLock.Unlock()
 }
 
 func (l statusLogger) OnTaskProgress(ctx context.Context, t *Task) {
+	l.lastProgressUpdateLock.Lock()
 	if time.Since(l.lastProgressUpdate[t]) > l.progressUpdateFreq {
-		log.I(ctx, "%v (%v%%) %v", t, t.Completion(), t.TimeSinceStart())
 		l.lastProgressUpdate[t] = time.Now()
+		l.lastProgressUpdateLock.Unlock()
```

Comment:
Might be safer to unlock once at the end of the function and have a single return point?

Reply:
Done.